### ISSUE TITLE
how to set site base when deploy to github page?

### ISSUE THREAD

**Issue by jinhuan138** [SveltePress/sveltepress#159]
### Describe the bug

```js
kit: {
    adapter: adapter({
        pages: 'dist',
        fallback: '404.html',
    })
},
    paths: {
        base: '/base',
    }
},
```
build error :Error: 404 / does not begin with `base`,which is configured in `paths.base` and can be imported from `$app/paths`

### Reproduction

Is the problem svelte-kit or vite? 

### Severity

annoyance

**Comment by Blackman99** [comment]
Can I see your github link please?

**Comment by Blackman99** [comment]
Please create an empty .nojekyll file in your static folder. 
Reference here: https://kit.svelte.dev/docs/adapter-static#github-pages

**Comment by jinhuan138** [comment]
> Please create an empty .nojekyll file in your static folder. Reference here: https://kit.svelte.dev/docs/adapter-static#github-pages
> 
> I've forked your repo and tried it. It is working as expected: https://blackman99.github.io/svelte-reader-docs/

thanks a lot

**Comment by Hejtmus** [comment]
Hey @Blackman99 , 

thanks for this amazing lib. I encountered the same problem.

See this page: https://genoacms.github.io/docs/
This is the original repo: https://github.com/GenoaCMS/docs

When deploying the above mentioned way, it seemingly works, but whenever is user routed to other page of documentation, user gets 404. Also the images are not being loaded. Since the page in this issue lacks both, multiple docs pages and images, the problem can't be observed. But as you navigate through my docs page, you can see the effects.

I even tried setting the base path to match github's, however I had to prepend every link within my docs and that broke local run (npm run dev). Obviously if I don't add the basepath, my builds fail. I am using GH workflows for deploy.

Potential solution would be setting conditionally base path for all Sveltepress links. Is that possible, or am I doing something entirely wrong?

Best regards,
Filip Holcik.

**Comment by Blackman99** [comment]
See https://svelte.dev/docs/kit/$app-paths#base

**Comment by Hejtmus** [comment]
@Blackman99 
Yes, I know it, but whenever I override the base path, I must prepend all my links and images with the base (also in the sveltepress config) which is highly inconvenient. In case I don't do that, the build fails when prerendering:

```
node:internal/event_target:1099
  process.nextTick(() => { throw err; });
                           ^
Error: 404 /sail.png does not begin with `base`, which is configured in `paths.base` and can be imported from `$app/paths` - 
see https://svelte.dev/docs/kit/configuration#paths for more info (linked from /docs/)
```

Since the image in question is located in 'vite.config.ts', from `$app/paths` cannot be imported. I was reading code of this tool, and I found that some portions of code take base path into account, but some not.

Like this one:

- to property differentiates base paths: https://github.com/SveltePress/sveltepress/blob/b69716b3803673d65799869a0214773cb38d6e84/packages/theme-default/src/components/Logo.svelte#L8
- but when sourcing image, there is take value as is from the config, which results into not loaded image: https://github.com/SveltePress/sveltepress/blob/b69716b3803673d65799869a0214773cb38d6e84/packages/theme-default/src/components/Logo.svelte#L13

This is the result:

![Image](https://github.com/user-attachments/assets/ca2fc0a7-1574-4c98-b07e-70fd091dba86)

![Image](https://github.com/user-attachments/assets/dfd8ec84-3b04-4d80-8c19-332cf769d921)

see the href and src, href is correct, but the src lacks base path prefix. This is just one example, there are multiple instances of this character. I am willing to a create PR to fix this.

**Comment by Blackman99** [comment]
I can see your concern. 
Here's the explanation:
* It's hard to figure out what image the user pass
for example, assuming the site base is `/foo`, the passed image might be these values:
	- `/foo/bar.png`
	- `/bar.png`
	- `https://other.static.site/bar.png`
	- `data:image/png;base64,...`
* Logo component auto add `base` to root page https://github.com/SveltePress/sveltepress/blob/b69716b3803673d65799869a0214773cb38d6e84/packages/theme-default/src/components/Logo.svelte#L8
This link has no doubt always is *the root page* of the project. So it can easily auto determined no matter what the `base` is.

**Comment by Hejtmus** [comment]
I see the problem when resolving image, but in current state the image is not correctly resolved with custom base paths, which has simple workaround, prepending it in config. However next problems pop up after that.

https://github.com/SveltePress/sveltepress/blob/b69716b3803673d65799869a0214773cb38d6e84/packages/theme-default/src/components/SidebarGroup.svelte#L65

When I customize my base path, all my sidebar links are broken (and the bottom navigation as well), which could be mitigated by same workaround as the image, but repeating yourself in the config is not convenient.

I tried editing the Sveltepress components in node_modules, and the changes resolved this problem, however as you said, we can't distinguish which image src is user referring to, whether the first directory is directory or base path. One potential solution would be changing the format and explicitly prepending the image with the base path, and mentioning that in docs. Question is how many users will this change affect. 

For now it seems like I will have to choose something different from GH pages, some environment where I won't have to customize base path. Or make a fork with these changes, since even if you agree to these changes, we would have to wait until major release.

**Comment by Blackman99** [comment]
GH pages can totally work. As you mentioned:
>  repeating yourself in the config

Sometimes repeatings can be recognized as a trade-off between exlicit and ambiguous. 

But indeed the most common case for sidebar is to auto add `base` prefix. 
So maybe it is a better idea to make it auto add by default. And preserve the ability to disable it.

Thanks for your feedback!

**Comment by Hejtmus** [comment]
Hey @Blackman99,

thanks for your time and effort when resolving this issue. However I would like to ask you to release the patch/minor version as I plan using Sveltepress with the changes. 
BTW neatly structured repo, I really enjoyed working with it, and quality of your code is really high.

Have a nice day!

**Comment by Blackman99** [comment]
New version has been released.

I'll leave this issue open for a while in case of some unexpected problems while using the new version.

**Comment by Blackman99** [comment]
Feel free to reopen this if there are any further problems
